server/container_create.go

```diff
@@ -351,6 +370,48 @@ func (s *Server) CreateContainer(ctx context.Context, req *pb.CreateContainerReq
 	return resp, nil
 }
 
+func (s *Server) setupOCIHooks(specgen *generate.Generator, sb *sandbox.Sandbox, containerConfig *pb.ContainerConfig, command string) error {
+	mounts := containerConfig.GetMounts()
+	for _, hook := range s.config.Hooks {
+		var (
+			match bool
+			err   error
+		)
+		logrus.Debugf("SetupOCIHooks", hook)
+		if hook.HasBindMounts && len(mounts) > 0 {
+			if err := addOCIHook(specgen, hook); err != nil {
+				return err
+			}
+			continue
+		}
+		for _, cmd := range hook.Cmds {
+			if match, err = regexp.MatchString(cmd, command); err != nil {
```

Comment:
Nit: can you make match scoped only to the loop? Just to make sure it doesn't accidentally get shared between iterations of the loop if we make this more complicated

Reply:
Fixed